File: app/src/lumen/java/com/cliqz/browser/controlcenter/ControlCenterViewPager.java
package com.cliqz.browser.controlcenter;

import android.content.Context;
import android.util.AttributeSet;

import androidx.viewpager.widget.ViewPager;

/**
 * Copyright © Cliqz 2019
 */
public class ControlCenterViewPager extends ViewPager {

    public ControlCenterViewPager(Context context) {
        this(context, null);
    }

    public ControlCenterViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
        addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                // Do nothing
            }

            @Override
            public void onPageSelected(int position) {
                if (getAdapter() != null) {
                    ((ControlCenterPagerAdapter) getAdapter()).updateCurrentView(position);
                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {
                // Do nothing
            }
        });
    }
}

File: app/src/lumen/java/com/cliqz/browser/controlcenter/ControlCenterViewPager.kt
package com.cliqz.browser.controlcenter

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent

import androidx.viewpager.widget.ViewPager

/**
 * Copyright © Cliqz 2019
 */
class ControlCenterViewPager @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null) : ViewPager(context, attrs) {

    @JvmField
    internal var isPagingEnabled = true

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        return this.isPagingEnabled && super.onTouchEvent(event)
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        return this.isPagingEnabled && super.onInterceptTouchEvent(event)
    }

}
